src/client/app/components/AdminComponent.jsx

```diff
@@ -3,15 +3,129 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 import React from 'react';
+import axios from 'axios';
+import { FormControl, Button } from 'react-bootstrap';
+import { chartTypes } from '../reducers/graph';
+import HeaderContainer from '../containers/HeaderContainer';
 
-/**
- * React component that controls the Admin panel
- * @returns JSX to create the AdminComponent
- */
-export default function AdminComponent() {
-	return (
-		<div>
-			<p>Admin panel</p>
-		</div>
-	);
+export default class AdminComponent extends React.Component {
+	constructor(props) {
+		super(props);
+		this.handleDisplayTitleChange = this.handleDisplayTitleChange.bind(this);
+		this.handleDefaultChartToRenderChange = this.handleDefaultChartToRenderChange.bind(this);
+		this.handleDefaultBarStackingChange = this.handleDefaultBarStackingChange.bind(this);
+		this.handleSubmitPreferences = this.handleSubmitPreferences.bind(this);
+	}
+
+	handleDisplayTitleChange(e) {
+		this.props.updateDisplayTitle(e.target.value);
+	}
+
+	handleDefaultChartToRenderChange(e) {
+		this.props.updateDefaultGraphType(e.target.value);
+	}
+
+	handleDefaultBarStackingChange() {
+		this.props.toggleDefaultBarStacking();
+	}
+
+	handleSubmitPreferences() {
+		axios.post('/api/preferences',
```

Comment:
As far as I know, this is the only place in the project where we make a request from a react component.
Is there a very good reason for this?
This seems to make it difficult, if not impossible, to properly track whether or not the admin preferences have been submitted. I think this should be moved into an action, and the logic handling submissions made significantly more robust.

Reply:
There are multiple places this is done, including LoginComponent, AdminComponent, CreateGroup component, etc. It could be moved into Redux, but it ultimately does not touch state at all. Therefore, there is no reason to put it into Redux. Note that the examples above are all POST requests.

Reply:
This is something I'd like to fix eventually. We should refactor all of the network stuff out into its own part of the project and then only call it from thunk actions.

Reply:
@johndiiorio of the examples you mentioned, in master, only loginComponent makes a post request from the react.* That example is fundamentally different because it involves handling passwords.
This should move to redux. It will need to touch state to toggle a `submitted` flag. 

*If there are versions of these that _do_ make post requests from the react, then I will most likely have exactly the same problems with them.

Reply:
I agree with Simon; I think once TypeScript is done (soon!) we can do this pretty easily.

Reply:
My philosophy is that Redux should only be for the code that directly interacts with the state. If we want to factor out all the network requests into some file, that's fine. I just disagree with the all Network requests => places in thunks philosophy.

Also, @jameeters, I don't see the need of a submitted flag for submitting admin preferences. Could you elaborate as to why that is necessary?